### PR TITLE
chore(I3S & I3SDebug): UI improvements

### PR DESCRIPTION
**Pull request by morphoix**
debug-panel
debug-dark

### CODE REVIEW

**Review by ibgreen — APPROVED** [review]
Very cool sci-fi look!

File: examples/website/i3s/components/debug-panel.js [hunk 1]
```diff
@@ -1,88 +1,106 @@
...
 
 const Header = styled.h3`
   margin: 0;
+  padding: 5px;
+  color: #00ADE6;
+  text-shadow: 1px 1px 1px #212529;
+  text-align: center;
+  top: 0px;
+  position: -webkit-sticky;
+  position: sticky;
```
Comment (belom88) [inline examples/website/i3s/components/debug-panel.js:50]: `sticky` is not supported in IE

File: examples/website/i3s/components/debug-panel.js [hunk 2]
```diff
@@ -1,88 +1,106 @@
...
+    color: white;
+  }
 `;
 
 const ChildWrapper = styled.div`
   margin-top: 10px;
 `;
 
-const Label = styled.label`
+export const Label = styled.label`
```
Comment (belom88) [inline examples/website/i3s/components/debug-panel.js:96]: Why `export`?

File: examples/website/i3s/components/memory-stats.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  z-index: 1;
+  margin: 5px;
+  color: #adb5bd;
+  background: rgba( 0,0,0,.7);
+  text-transform: uppercase;
+  border: 1px solid rgb(52, 152, 219);
+  align-items: center;
+  height: 28px;
+  cursor: pointer;
+  border: 2px solid #212529;
```
Comment (belom88) [inline examples/website/i3s/components/memory-stats.js:39]: Too many `border`

File: examples/website/i3s/app-debug.js [hunk 4]
```diff
@@ -504,14 +505,13 @@ export default class App extends PureComponent {
...
       >
-        {this._renderStats()}
       </DebugPanel>
     );
   }
 
   _renderStats() {
     // TODO - too verbose, get more default styling from stats widget?
-    return <div style={STATS_WIDGET_STYLE} ref={(_) => (this._statsWidgetContainer = _)} />;
+    return <StatsWidgetContainer ref={(_) => (this._statsWidgetContainer = _)} />;
```
Comment (belom88) [inline examples/website/i3s/app-debug.js:514]: Remove `const STATS_WIDGET_STYLE = {`

File: examples/website/i3s/app-debug.js [hunk 5]
```diff
@@ -719,13 +719,19 @@ export default class App extends PureComponent {
 
   render() {
     const layers = this._renderLayers();
-    const {selectedMapStyle, tileInfo, debugOptions} = this.state;
+    const {selectedMapStyle, tileInfo, debugOptions, showMemory} = this.state;
```
Comment (belom88) [inline examples/website/i3s/app-debug.js:722]: init `showMemory` in constructor

File: examples/website/i3s/app-debug.js [hunk 6]
```diff
@@ -719,13 +719,19 @@ export default class App extends PureComponent {
...
 
     return (
       <div style={{position: 'relative', height: '100%'}}>
         {this._renderDebugPanel()}
         {tileInfo ? this._renderAttributesPanel() : this._renderControlPanel()}
         {debugOptions.semanticValidator && this._renderSemanticValidator()}
+        <MemoryButton onClick={() => this.setState({showMemory: !showMemory})}>
+          {showMemory ? `Hide ` : `Show `}Memory Usage
```
Comment (belom88) [inline examples/website/i3s/app-debug.js:730]:
```suggestion
          {showMemory ? `Hide` : `Show`} Memory Usage
```

File: examples/website/i3s/app.js [hunk 7]
```diff
@@ -229,11 +213,14 @@ export default class App extends PureComponent {
 
   render() {
     const layers = this._renderLayers();
-    const {viewState, selectedMapStyle, selectedFeatureAttributes} = this.state;
+    const {viewState, selectedMapStyle, selectedFeatureAttributes, showMemory} = this.state;
```
Comment (belom88) [inline examples/website/i3s/app.js:216]: init `showMemory` in constructor

File: examples/website/i3s/app.js [hunk 8]
```diff
@@ -229,11 +213,14 @@ export default class App extends PureComponent {
...
 
     return (
       <div style={{position: 'relative', height: '100%'}}>
-        <StatsWidgetWrapper>{this._renderStats()}</StatsWidgetWrapper>
+        <MemoryButton onClick={() => this.setState({showMemory: !showMemory})}>{showMemory ? `Hide ` : `Show `}Memory Usage</MemoryButton>
```
Comment (belom88) [inline examples/website/i3s/app.js:220]: Ditto

File: examples/website/i3s/components/control-panel.js [hunk 9]
```diff
@@ -8,54 +8,113 @@ const Container = styled.div`
...
   font-weight: 800;
-  font-size: 16px;
+  cursor: pointer;
+  margin: 5px;
+  font-size: 13px;
+  text-transform: uppercase;
+  color: #00ADE6;
+  text-shadow: 1px 1px 1px #212529;
+  background: rgba( 0, 0, 0, .3);
+  background: transparent;
```
Comment (belom88) [inline examples/website/i3s/components/control-panel.js:52]: Too many `background`

**Review by belom88 — CHANGES_REQUESTED** [review]

File: examples/website/i3s/components/debug-panel.js [hunk 10]
```diff
@@ -1,88 +1,106 @@
...
+  font-size: 11px;
+  jusify-content: center;
 `;
 
 const Expander = styled.div`
   top: calc(50% - 54px);
   width: 14px;
   padding: 10px 0px 10px 1px;
-  background: #fff;
+  background: rgba( 36, 39, 48, 0.7);
```
[the remark this answers is not in the document]
Comment (morphoix) [reply examples/website/i3s/components/debug-panel.js:75]: the rule opacity will make the whole element transparent but I need only background to be transparent

File: examples/website/i3s/components/control-panel.js [hunk 11]
```diff
@@ -8,54 +8,113 @@ const Container = styled.div`
...
   align-items: center;
-  height: 20px;
+  justify-content: center;
+  border-right: 2px solid #212529;
+  -webkit-transition: all 1s ease;
+  -moz-transition: all 1s ease;
+  -o-transition: all 1s ease;
+  transition: all 1s ease;
+  &:hover {
+    color: rgb(36,39,48);
```
[the remark this answers is not in the document]
Comment (morphoix) [reply examples/website/i3s/components/control-panel.js]: > convert to HEX to have all colors in the same notation

fixed everywhere except cases with alpha-channel

File: examples/website/i3s/components/debug-panel.js [hunk 12]
```diff
@@ -119,24 +136,28 @@ export default class DebugPanel extends PureComponent {
...
     };
   }
 
   _clearButtonStyles(isClearButtonDisabled) {
     return {
       display: 'flex',
-      color: 'white',
-      background: isClearButtonDisabled ? 'gray' : 'red',
+      background: isClearButtonDisabled ? '#212529' : '#00ADE6',
+      color: isClearButtonDisabled ? '#f2e9e4' : 'rgb(36,39,48)',
```
Comment (belom88) [inline examples/website/i3s/components/debug-panel.js:153]: use HEX, please

File: examples/website/i3s/components/memory-stats.js [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+  text-transform: uppercase;
+  align-items: center;
+  height: 28px;
+  cursor: pointer;
+  border: 2px solid #212529;
+  border-top: none;
+  @media screen and (max-width: 768px) {
+   bottom: 20px;
+  }
+`;
```
Comment (belom88) [inline examples/website/i3s/components/memory-stats.js:43]: New line in the end of the file

File: examples/website/i3s/components/tile-validator.js [hunk 14]
```diff
@@ -35,6 +39,7 @@ const NormalsValidator = styled.div`
 const ValidatorInfoList = styled.div`
   display: flex;
   flex-direction: column;
+  background: gba( 0,0,0,.5);
```
Comment (belom88) [inline examples/website/i3s/components/tile-validator.js:42]: `rgba`?

File: examples/website/i3s/components/control-panel.js [hunk 15]
```diff
@@ -8,54 +8,113 @@ const Container = styled.div`
...
+  line-height: 2;
+  text-transform: uppercase;
   align-items: center;
-  height: 20px;
+  justify-content: center;
+  border-right: 2px solid #212529;
+  -webkit-transition: all 1s ease;
+  -moz-transition: all 1s ease;
+  -o-transition: all 1s ease;
+  transition: all 1s ease;
```
[the remark this answers is not in the document]
Comment (belom88) [reply examples/website/i3s/components/control-panel.js:87]: Not fixed

File: examples/website/i3s/components/semantic-validator.js [hunk 16]
```diff
@@ -56,9 +66,9 @@ export default class SemanticValidator extends PureComponent {
   renderColumns(warnings) {
     return warnings.map((warning, index) => (
       <tr key={`${warning.title}-${index}`}>
-        <td style={{padding: '3px'}}>{index + 1}</td>
-        <td style={{padding: '3px'}}>{WARNING_TYPES[warning.type]}</td>
-        <td style={{color: 'red', padding: '3px'}}>{warning.title}</td>
+        <td style={{padding: '3px', color: 'rgb(0, 173, 230)'}}>{index + 1}</td>
```
Comment (belom88) [inline examples/website/i3s/components/semantic-validator.js:69]: Use HEX

File: examples/website/i3s/components/semantic-validator.js [hunk 17]
```diff
@@ -56,9 +66,9 @@ export default class SemanticValidator extends PureComponent {
...
+        <td style={{padding: '3px', color: 'rgb(242, 233, 228)'}}>{WARNING_TYPES[warning.type]}</td>
```
Comment (belom88) [inline examples/website/i3s/components/semantic-validator.js:70]: Use HEX

File: examples/website/i3s/components/toggle-switch.js [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+const Input = styled.input`
+  height: 0;
+  width: 0;
+  opacity: 0;
+  z-index: -1;
+`;
+
+const Label = styled.label`
+  position: absolute;
+  display: inline-block;
```
Comment (belom88) [inline examples/website/i3s/components/toggle-switch.js:13]: `display: inline-block` and `position: absolute` simultaneously doesn't have sense.

File: examples/website/i3s/components/toggle-switch.js [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+
+const Slider = styled.span`
+  position: absolute;
+  cursor: pointer;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  background-color: rgb(36, 39, 48);
+  -webkit-transition: 0.4s;
```
Comment (belom88) [inline examples/website/i3s/components/toggle-switch.js:35]: if you want to use browser-specific extensions for `transition`, use it everywhere in your code.

File: examples/website/i3s/components/toggle-switch.js [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+
+  &::after {
+    color: white;
+  }
+  &::before {
+    position: absolute;
+    content: "";
+    height: 1.5em;
+    width: 2em;
+    box-shadow: rgb(0 0 0 / 40%) 0px 2px 4px 0px;
```
Comment (belom88) [inline examples/website/i3s/components/toggle-switch.js:47]: Don't see effect of this rule on the screen

File: examples/website/i3s/components/toggle-switch.js [hunk 21]
```diff
@@ -0,0 +1,85 @@
...
+    background-color: #f2e9e4;
+    -webkit-transition: 0.4s;
```
Comment (belom88) [inline examples/website/i3s/components/toggle-switch.js:49]: Ditto

**Review by belom88 — CHANGES_REQUESTED** [review]
"Show more/less info" animation doesn't look correct;
"Statistics" overlaps "Debug Panel". We need to be able to use these panels simultaneously
"Clear all" look different in Firefox;
Select options look different in Firefox;

File: examples/website/i3s/components/debug-panel.js [hunk 22]
```diff
@@ -1,76 +1,91 @@
...
+  @media screen and (max-width: 768px) {
+   margin: 0;
+  };
 `;
 
 const Header = styled.h3`
   margin: 0;
+  padding: 5px;
+  color: #00ADE6;
+  text-shadow: 1px 1px 1px #212529;
```
[the remark this answers is not in the document]
Comment (morphoix) [reply examples/website/i3s/components/debug-panel.js:46]: > Don't see effect of this line

the text-shadow is for light map

**Review by belom88 — APPROVED** [review]

**Pull request merged**